src/Persistence.LinnApps/Repositories/BomRepository.cs

```diff
@@ -45,7 +45,7 @@ public Bom FindBy(Expression<Func<Bom, bool>> expression)
         {
             return this.serviceDbContext.Boms.Where(expression)
                 .Include(b => b.Details)
-                .ThenInclude(d => d.Part)
+                .ThenInclude(d => d.Part).ThenInclude(p => p.PartSuppliers.Where(ps => ps.SupplierRanking == 1))
```

Comment:
is supplier ranking ==1more reliable that parts - > part_supplier via preferred_supplier?